lib/devise/controllers/sign_in_out.rb

```diff
@@ -51,6 +51,7 @@ def sign_in(resource_or_scope, *args)
           true
         else
           warden.set_user(resource, options.merge!(scope: scope))
+          resource.update_tracked_fields!(warden.request) if update_tracked_fields?(resource, warden)
```

Comment:
The `#update_tracked_fields?` logic is being duplicated here and inside the `database_authenticatable` strategy. My concern is that in the future if there's a need to change this logic one of the places could be forgotten, possibly causing issues.
I'd say to move this conditional to `Devise::Models::Trackable#update_tracked_fields!`. We'll get the benefit of keeping the logic in the same place and also the caller wouldn't need to know about that conditional - i.e. "tell don't ask".
WDYT?